contracts/dollar/tasks/priceReset.ts

```diff
@@ -11,70 +11,41 @@ import { IMetaPool } from "../artifacts/types/IMetaPool";
 import { BondingV2 } from "../artifacts/types/BondingV2";
 import { ERC20 } from "../artifacts/types/ERC20";
 import pressAnyKey from "../utils/flow";
-import { TWAPOracle } from "../artifacts/types/TWAPOracle";
-import { EthereumProvider } from "hardhat/types";
-import { JsonRpcProvider } from "@ethersproject/providers";
+import { DEPLOYMENT_OVERRIDES, FORKING_CHAIN_ID } from "./constants"
+import { A_PRECISION, get_burn_lp_amount } from "./utils"
 
 dotenv.config();
 
-const {
-  API_KEY_ALCHEMY
-} = process.env;
-
 task(
   "priceReset",
   "PriceReset can push uAD price lower or higher by burning LP token for uAD or 3CRV from the bonding contract"
 )
-  .addParam("amount", "The amount of uAD-3CRV LP token to be withdrawn")
-  .addOptionalParam(
-    "pushhigher",
-    "if false will withdraw 3CRV to push uAD price lower",
-    true,
-    types.boolean
-  )
+  .addParam("price", "The target price of uAD")
   .addOptionalParam(
     "dryrun",
     "if false will use account 0 to execute price reset",
     true,
     types.boolean
   )
-  .addOptionalParam(
-    "blockheight",
-    "block height for the fork",
-    14800000,
-    types.int
-  )
   .setAction(
     async (
       taskArgs: {
-        amount: string;
-        pushhigher: boolean;
+        price: number;
         dryrun: boolean;
-        blockheight: number;
+        twapUpdate: boolean
       },
-      { ethers, network, getNamedAccounts }
+      { ethers, network, getNamedAccounts, deployments }
     ) => {
-      const net = await ethers.provider.getNetwork();
-      const resetFork = async (blockNumber: number): Promise<void> => {
-        await network.provider.request({
-          method: "hardhat_reset",
-          params: [
-            {
-              forking: {
-                jsonRpcUrl: `https://eth-mainnet.alchemyapi.io/v2/${API_KEY_ALCHEMY || ""
-                  }`,
-                blockNumber,
-              },
-            },
-          ],
-        });
-      };
+      console.log("started....");
+      const { chainId } = network.config;
+      const OVERRIDES_PARAMS = DEPLOYMENT_OVERRIDES[chainId!];
```

Comment:
There might be some cases we have to use already deployed contract addresses in hh tasks. you were doing so. but you put  some contract addresses in namedAccounts of hardhat.config.ts which looks not much extensible to me. 
For example, lets assume curveBasePool's address is Addr1 on rinkeby, Addr2 on goerli. I thought that would be better if we handle the deployed contract addresses per chain. 
so I kept it (because it was being used in other functions) and copied it to constants/index.ts.  

sorry for not commenting well.